### PR TITLE
Fix broken Storybook coverage for setup.

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

Addresses issue #3074

## Relevant technical choices

* Passes necessary datastore configuration to the registry so that the stories work as expected.

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: stories/setup.stories.js [hunk 1]
```diff
@@ -27,94 +27,130 @@ import { storiesOf } from '@storybook/react';
...
+		],
 	} )
-	.add( 'Disconnected - URL Mismatch [User Input]', () => {
-		const setupRegistry = ( registry ) => {
-			provideUserAuthentication( registry, {
-				authenticated: false,
-				disconnectedReason: DISCONNECTED_REASON_CONNECTED_URL_MISMATCH,
-			} );
-		};
+	.add( 'Disconnected - URL Mismatch [User Input]', ( args, { registry } ) => {
```
Comment (aaemnnosttv) [inline stories/setup.stories.js:137]: These two look the same as the stories for "Start" and "Start [User Input]" – is that expected?

File: stories/setup.stories.js [hunk 2]
```diff
@@ -27,94 +27,130 @@ import { storiesOf } from '@storybook/react';
...
-				authenticated: false,
-				requiredScopes: [],
-				grantedScopes: [],
-			} );
-
-			dispatch( CORE_USER ).receiveGetTracking( {
-				enabled: false,
-			} );
-		};
+		registry.dispatch( CORE_USER ).receiveGetAuthentication( {
```
Comment (aaemnnosttv) [inline stories/setup.stories.js:49]: Should we use `provideUserAuthentication` instead? If not, we should probably annotate why here.

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @felixarntz ! Looks good for the most part, just a few questions.

File: stories/setup.stories.js (hunk 1 above)
Comment (felixarntz) [reply stories/setup.stories.js]: Just pushed commits to address that :)

**Review by aaemnnosttv — APPROVED** [review]
LGTM 👍

**Pull request merged**